walker.go

```diff
@@ -219,10 +219,10 @@ func trimUntilIndex(str, sep string, count int) string {
 }
 
 func isNotExist(err error) bool {
-	err = errors.Cause(err)
-	if os.IsNotExist(err) {
+	if errors.Is(err, os.ErrNotExist) || errors.Is(err, syscall.ENOTDIR) {
 		return true
 	}
+	err = errors.Cause(err)
 	if pe, ok := err.(*os.PathError); ok {
 		err = pe.Err
 	}
```

Comment:
Wondering if this code is still needed,  or if the `errors.Is(err, syscall.ENOTDIR)` I added above captures this already

@AkihiroSuda @tonistiigi ?

Reply:
This should use `errors.As` instead of `Cause`

Reply:
Didn't see the line comparing to `syscall.ENOTDIR` below. I guess this part isn't needed at all then now.

Reply:
Thanks! Yes, I was in doubt if `errors.Is()` would do everything that type-matching `err.(*os.PathError)` did. I'll remove the part